### ISSUE TITLE
Change Request: Support `applyInlineConfig`

### ISSUE THREAD

**Issue by lumirlumir** [eslint/markdown#330]
### Environment

ESLint version: `v9.23.0`
@eslint/markdown version: `v6.3.0`
Node version: `v20.18.0`
npm version: `v10.9.2`
Operating System: Windows 11

### What problem do you want to solve?

Hello,  

Currently, unlike JSON and JavaScript parsers, Markdown parsers do not recognize inline configurations like the following:  

```js  
/* eslint eqeqeq: "off", curly: "error" */  
```  

Therefore, I would like to propose adding support for `applyInlineConfig` to `MarkdownSourceCode`.

### What do you think is the correct solution?

Here is the implemented version for the JSON language:  

https://github.com/eslint/json/blob/main/src/languages/json-source-code.js#L208-L250

Let's create an `applyInlineConfig` for Markdown language as well.

### Participation

- [x] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
What would you propose the syntax look like for Markdown? We can't use C-style comments for this.

**Comment by lumirlumir** [comment]
@nzakas  

I would like to suggest using the HTML-style comment syntax, which has been used to enable or disable ESLint in Markdown.  

Just as we use `<!-- eslint-disable -->` to disable ESLint linting in Markdown, I believe we can adopt a similar syntax, like this:  

```md  
<!-- eslint no-html: "true" -->  
```

**Comment by nzakas** [comment]
That sounds good to me. :+1: Let's do it!
